dsgrid/dimension/dimension_filters.py

```diff
@@ -0,0 +1,230 @@
+
+        Returns
+        -------
+        str
+
+        """
+
+
+class DimensionFilterExpressionModel(_DimensionFilterWithWhereClauseModel):
+    """Filters a table where a dimension column matches an expression.
+
+    Example:
+        DimensionFilterExpressionModel(
+            dimension_type=DimensionType.GEOGRAPHY,
+            dimension_query_name="county",
+            operator="==",
+            value="06037",
+        ),
+    is equivalent to
+        df.filter("county == '06037'")
+
+    """
+
+    operator: str
+    value: Union[str, int, float]
+
+    def where_clause(self, column=None):
+        column = column or self.column
+        value = self._make_value_str(self.value)
+        text = f"({column} {self.operator} {value})"
+        return text
+
+
+class DimensionFilterExpressionRawModel(_DimensionFilterWithWhereClauseModel):
+    """Filters a table where a dimension column matches an expression.
+    Uses the passed string with no modification.
+
+    Example:
+        DimensionFilterExpressionModel(
+            dimension_type=DimensionType.GEOGRAPHY,
+            dimension_query_name="county",
+            value="== '06037'",
+        ),
+    is equivalent to
+        df.filter("county == '06037'")
+
+    The difference between this class and DimensionFilterExpressionModel is that the latter
+    will attempt to add quotes as necessary.
+
+    """
+
+    value: Union[str, int, float]
+
+    def where_clause(self, column=None):
+        column = column or self.column
+        text = f"({column} {self.value})"
+        return text
+
+
+DIMENSION_COLUMN_FILTER_OPERATORS = {
+    "between",
+    "contains",
+    "endswith",
+    "isNotNull",
+    "isNull",
+    "isin",
+    "like",
+    "rlike",
+    "startswith",
+}
+
+
+def check_operator(operator):
+    if operator not in DIMENSION_COLUMN_FILTER_OPERATORS:
+        raise ValueError(
+            f"operator={operator} is not supported. Allowed={DIMENSION_COLUMN_FILTER_OPERATORS}"
+        )
+    return operator
+
+
+class DimensionFilterColumnOperatorModel(DimensionFilterBaseModel):
+    """Filters a table where a dimension column matches a Spark SQL operator.
+
+    Examples:
+    import pyspark.sql.functions as F
+    df.filter(F.col("geography").like("abc%"))
+    df.filter(~F.col("sector").startswith("com"))
+    df.filter(F.col("timestamp").between("2012-07-01 00:00:00", "2012-08-01 00:00:00"))
+    """
+
+    value: Any = Field(title="value", description="Value to filter on")
+    operator: str = Field(
+        title="operator", description="Method on pyspark.sql.functions.col to invoke"
+    )
+    negate: bool = Field(
+        title="negate",
+        description="Change the filter to match the negation of the value.",
+        default=False,
+    )
+
+    @validator("operator")
+    def check_operator(cls, operator):
+        return check_operator(operator)
+
+    def apply_filter(self, df, column=None):
```

Comment:
Have you tested this method? I feel like it's missing sth (e.g., where do you specify the column/dimension)? Or I just need more documentation.

Reply:
The `between` case needs to cover the case where `value` is a list. Current answer: not handled because we don't support this yet.